frontend/packages/console-dynamic-plugin-sdk/src/extensions/storage-class-provisioner.ts

```diff
@@ -0,0 +1,39 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { CodeRef, ExtensionDeclaration } from '../types';
+
+export type ProvisionerProps = {
+  onParamChange: (id: string) => void;
+};
+
+export type StorageClassProvisioner = ExtensionDeclaration<
+  'console.storage-class/provisioner',
+  {
+    [key: string]: {
+      [key: string]: {
+        title: string;
+        provisioner: string;
+        allowVolumeExpansion: boolean | CodeRef<(arg) => boolean>;
```

Comment:
Why not simply `CodeRef<(arg) => boolean>` ?

I'm not strictly against these `value | CodeRef<func_to_get_value>` unions, it's just that we need to check first, if there aren't any issues from generated JSON schema, build and runtime perspectives.